### PR TITLE
Update starter code for schema2qa and add documentation

### LINKED ISSUE #273: Finish writing up the starter tutorials
We spent a lot of effort to set up the "starters" in a way that people can get something done quickly with Genie, but there is no documentation, so people end up using the outdated CS294S homeworks (see e.g. https://github.com/stanford-oval/cs294-homework-workdir/issues/2)
This is not good. We need to document the new, always up to date, starter directories and point newcomers that way.

### PR DESCRIPTION
**Pull request by rayslxu**
Fix #273

### CODE REVIEW

File: starter/schemaorg/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+After installation, you should get a command called `thingpedia`.
+If encounter `command not found`, make sure the Yarn global bin directory
+(usually `~/.yarn/bin`) is in your PATH. You can find the path with the command
+`yarn global bin`.
+
+```bash
+export PATH=~/.yarn/bin:$PATH
+```
```
Comment (gcampax) [inline starter/schemaorg/README.md:35]: This installation bit doesn't seem right. We already have installation instructions for Genie in doc/install.md, and also the starter is literally inside the Genie repository so one must have cloned the repository first.

File: starter/schemaorg/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+## Configuration
+
+Edit `Makefile` and set `geniedir` with the path to your "genie-toolkit"
+checkout. Set `developer_key` to your Thingpedia developer key.
```
Comment (gcampax) [inline starter/schemaorg/README.md:40]: Can we make geniedir default to `../..` ?

Also, someone was asking today how to get a developer key (about a different guide). We should probably link to a page that explains how to get a developer key, or link to the "Apply to be a developer page" (on the correct server!)

File: starter/schemaorg/README.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+However, different websites for the same domain might use a different set of properties, and provide
+data in different format. 
+Thus, Schema2QA replies on real Schema.org metadata to decide the final schema, and generate training data.
+
+We have prepared source data for the 6 domains by crawling existing Schema.org metadata from 
+Yelp, Hyatt, LinkedIn, IMDb, Last.fm, and Goodreads, respectively. 
+
+
+### Generate Manifest and Value Datasets
+Manifest.tt file contains the manifest of the Q\&A agent to build, including all the queries and properties
```
Comment (gcampax) [inline starter/schemaorg/README.md:69]:
```suggestion
The manifest.tt file contains the manifest of the Q\&A agent to build, including all the queries and properties 
```

File: starter/schemaorg/README.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+The starter code includes evaluation data crowd sourced from Amazon MTruk for the existing
+6 domains. `multidomain` experiments contains the evaluation data for all 6 domains. 
+```bash
+make experiment=$(exp) model=$(model) evaluate
+```
+
+If you obtain your own , you can add it to `$(experiment)/eval/annotated.tsv` for the dev set,
+and `$(experiment)/test/annotated.tsv` for the test set.
+
+Finally, you can evaluate a specific with
```
Comment (gcampax) [inline starter/schemaorg/README.md:134]: What did you mean here? A specific model? A specific evaluation set?

File: starter/schemaorg/README.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+```
+make experiment=... eval_set=... evaluate
+```
+`eval_set` should be "eval" or "test".
+
+Data added to the dev set will be also used during training for cross-validation.
```
Comment (gcampax) [inline starter/schemaorg/README.md:140]: This line seems spurious? Or it should be moved to the paragraph that goes "you can add it ..."

File: starter/schemaorg/README.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+### Evaluate
+
+The starter code includes evaluation data crowd sourced from Amazon MTruk for the existing
+6 domains. `multidomain` experiments contains the evaluation data for all 6 domains. 
+```bash
+make experiment=$(exp) model=$(model) evaluate
+```
+
+If you obtain your own , you can add it to `$(experiment)/eval/annotated.tsv` for the dev set,
```
Comment (gcampax) [inline starter/schemaorg/README.md:131]:
```suggestion
If you obtain your own evaluation data, you can add it to `$(experiment)/eval/annotated.tsv` for the dev set,
```

File: starter/schemaorg/Makefile [hunk 7]
```diff
@@ -93,77 +91,80 @@ custom_train_nlu_flags ?=
...
-	$(genie) augment -o $@.tmp -l en-US --thingpedia $(experiment)/schema.tt --parameter-datasets $(experiment)/parameter-datasets.tsv \
+%/augmented.tsv : %/synthetic.tsv %/parameter-datasets.tsv
+	$(genie) augment -o $@.tmp -l en-US --thingpedia $*/manifest.tt --parameter-datasets $*/parameter-datasets.tsv \
 	  --synthetic-expand-factor 1 --quoted-paraphrasing-expand-factor 60 --no-quote-paraphrasing-expand-factor 20 --quoted-fraction 0.0 \
-	  --debug $($(experiment)_paraphrase) $(experiment)/synthetic.tsv
+	  --debug $($(*)_paraphrase) $*/synthetic.tsv
 	mv $@.tmp $@
 
+multidomain/augmented.tsv: $(addsuffix /augmented.tsv,$(all_experiments))
+	cat $^ > $@
```
Comment (gcampax) [inline starter/schemaorg/Makefile:153]: Are we missing the rule to make `multidomain/eval/annotated.tsv` from the other files?

**Comment by rayslxu** [comment]
@gcampax I think I addressed all the comments

**Review by gcampax — APPROVED** [review]
Yeah let's merge this.
